ftw/publisher/core/adapters/ftw_shop.py

```diff
@@ -122,3 +123,69 @@ def setData(self, data, metadata):
             variations_config.purge_dict()
         else:
             variations_config.updateVariationConfig(data)
+
+
+class ShopCategorizableReferences(object):
+    """DataCollector adapter for the category references of the categorizable shop types.
+    """
+
+    implements(IDataCollector)
+    logger = getLogger()
+    security = ClassSecurityInformation()
+
+    def __init__(self, obj):
+        self.object = obj
+
+    security.declarePrivate('getData')
+    def getData(self):
+        categories_uids = [reference.UID() for reference in self.object.listCategories()]
+        data = make_serializable(categories_uids)
```

Comment:
Unless the data contains custom types, I would no longer use `make_serializable` too much since the data returned by `getData` is optimized for serialization automatically by the publisher.
See:
- [`decode_for_json`](https://github.com/4teamwork/ftw.publisher.core/blob/master/ftw/publisher/core/utils.py#L123)
- [`encode_after_json`](https://github.com/4teamwork/ftw.publisher.core/blob/master/ftw/publisher/core/utils.py#L216)

This mechanism makes sure that if `getData` returns e.g. a `PersistentMapping`, `setData` will be called with a `PersistentMapping` too.
The `make_serializable` function here does not do that: all `dict` objects are converted to `PersistentMapping`, which is not accurate.

So I propose to not use the functions `make_serializable` function of the `ftw_shop.py` but rely on the `utils.py` functions for new code (but I would not change existing adapters at this point).